### PR TITLE
migrate from dep to go modules

### LINKED ISSUE #235: Refactor ibm cloud provider use go module
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

https://github.com/kubernetes/enhancements/blob/master/keps/sig-architecture/2019-03-19-go-modules.md
noticed this, should we follow this or keep using dep or take this a future action?

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by songleo**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #235 

**Specify your PR reviewers**:

/cc @gyliu513 @jichenjc @xunpan

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

    No image version change

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,11 +13,12 @@ GOBIN_DEFAULT := $(GOPATH)/bin
 export GOBIN ?= $(GOBIN_DEFAULT)
 TESTARGS_DEFAULT := "-v"
 export TESTARGS ?= $(TESTARGS_DEFAULT)
-PKG := $(shell awk  -F "\"" '/^ignored = / { print $$2 }' Gopkg.toml)
 DEST := $(GOPATH)/src/$(GIT_HOST)/$(BASE_DIR)
 SOURCES := $(shell find $(DEST) -name '*.go')
 
-HAS_DEP := $(shell command -v dep;)
+# Kctive module mode, as we use go modules to manage dependencies
```
Comment (gyliu513) [inline Makefile]: What is `Kctive`

Comment (songleo) [reply Makefile]: This is a typo,  should be `# Active module mode ...`

**Pull request merged**